src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -97,53 +103,125 @@ static bool simplify_concat(std::shared_ptr<Node> n)
             return false;
         }
 
-        if (goe)
+        if (branch_tip)
         {
-            if (goe != matcher->get_pattern_map()[lgoe])
+            if (branch_tip != matcher->get_pattern_map()[ltip])
             {
-                NGRAPH_DEBUG << goe->get_name() << " doesn't match "
-                             << matcher->get_pattern_map()[lgoe]->get_name();
+                NGRAPH_DEBUG << branch_tip->get_name() << " doesn't match "
+                             << matcher->get_pattern_map()[ltip]->get_name();
                 return false;
             }
         }
         else
         {
-            goe = matcher->get_pattern_map()[lgoe];
-            NGRAPH_DEBUG << "setting goe to " << goe->get_name();
+            branch_tip = matcher->get_pattern_map()[ltip];
+            NGRAPH_DEBUG << "setting branch_tip to " << branch_tip->get_name();
         }
 
-        auto it = carg;
-        while (it != goe)
+        auto slice = std::dynamic_pointer_cast<op::Slice>(matcher->get_pattern_map()[lslice]);
+
+        if (branch_tip)
         {
-            if (auto rcarg = std::dynamic_pointer_cast<op::Reshape>(carg))
+            auto cur_lower_bounds = slice->get_lower_bounds();
+            if (cur_lower_bounds < prev_lower_bounds)
```

Comment:
I guess the only bug vector here, albeit unlikely, is if one of the slices is left out and the included slices are all in the correct order wrt their lower_bounds.